executor/index_reader.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2017 PingCAP, Inc.
```

Comment:
Is it ok to put `index_reader` and `index_lookup` together ? It's confusing with `index_lookup` and `index_lookup_join`.

Reply:
reverted